rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -591,12 +599,19 @@ void SqliteStorage::fill_topics_and_types()
         std::string, std::string, std::string, std::string, std::string>();
 
       for (auto result : query_results) {
+        auto yaml_node = YAML::Load(std::get<3>(result));
+        auto decoded = YAML::decode_for_version<std::vector<rosbag2_storage::Rosbag2QoS>>(
+          yaml_node,
+          metadata_.version);
+        std::vector<rclcpp::QoS> offered_qos_profiles;
+        offered_qos_profiles.reserve(decoded.size());
+        std::copy(decoded.begin(), decoded.end(), std::back_inserter(offered_qos_profiles));
```

Comment:
Here, a conversion from serialized YAML (`std::string`) & `std::vector<rclcpp::QoS>` is needed.